src/meshes/t8code_mesh.jl

```diff
@@ -810,32 +985,32 @@ function count_interfaces(mesh::T8codeMesh)
     local_num_mpi_conform = 0
     local_num_mpi_mortars = 0
 
-    visited_global_mortar_ids = Set{UInt64}([])
+    visited_global_mortar_ids = Set{UInt128}([])
```

Comment:
Can you explain why it is necessary to use `UInt128` instead of `UInt64` as before?

Reply:
Sure! The associated computation like [this](https://github.com/trixi-framework/Trixi.jl/blob/611a5761efbdce07a3c20cbc444e603a511c4f2e/src/meshes/t8code_mesh.jl#L1072)
can lead to very big numbers since
```
max_level = 29
max_tree_num_elements = UInt128(2^ndims)^max_level
```
is very big. With meshes of a small number of `cmesh` elements respectively trees (around 10 trees) the computation fits into a `UInt64`. In case of a bigger number of trees you get an integer overflow. This is actually a serious bug which I fix with this PR. The bug came not into effect before since I tested only for smaller meshes with a small number of trees. While developing this PR this bug bubbled up. As a matter of fact, the code did not blow. It led to a subtle mismatching of MPI interfaces in case of MPI parallel execution. With `UInt128` there is more than enough space for extremely large numbers of trees.

Reply:
Great, thanks a lot!